src/cluster/Cluster.cc

```diff
@@ -298,3 +427,72 @@ int Cluster::from_xml(const string& xml)
 
     return 0;
 }
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void Cluster::set_error_message(const char *mod, const string& name, const string& message)
+{
+    static const int MAX_ERROR_MSG_LENGTH = 100;
+
+    SingleAttribute * attr;
+    ostringstream     error_value;
+
+    error_value << one_util::log_time() << ": " << message.substr(0, MAX_ERROR_MSG_LENGTH);
+
+    if (message.length() >= MAX_ERROR_MSG_LENGTH)
+    {
+        error_value << "... see more details in oned.log";
+
+        NebulaLog::error(mod, message);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html